packages/main/src/List.js

```diff
@@ -887,7 +902,7 @@ class List extends UI5Element {
 			return;
 		}
 
-		if (!this._selectionRequested && this.mode !== ListMode.Delete) {
+		if (!this._selectionRequested && this.mode !== ListMode.Delete && !this.noClickSelect) {
```

Comment:
This expression doesn't cover all cases. For example if we have SingleSelect mode where we don't have ui5-radio-button and ui5-checkbox selection is not possible.